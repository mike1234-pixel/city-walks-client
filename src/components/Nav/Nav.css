.nav-container {
  min-width: 100%; /* correct mdbreact max-width */
  padding: 0; /* correct mdbreact padding-left and padding-right of 15px*/
}

.nav-bar {
  background-color: var(--elegant-color-dark);
  color: white;
  box-shadow: none; /* remove mdbreact box-shadow*/
  font-family: "Comfortaa";
}

.logo {
  font-family: var(--logo-font);
  font-size: 150%;
}

@media (min-width: 768px) {
  .hamburger1 {
    visibility: hidden;
    display: none;
  }
}
