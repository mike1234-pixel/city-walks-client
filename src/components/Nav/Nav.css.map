{"version":3,"sourceRoot":"","sources":["Nav.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;AAEA;EACE;;;AAIJ;EACE;EACA;;;AAGF;EACE;IACE;IACA;;;AAKF;EACE;;AAEA;EAGE;;AAIJ;EACE;;;AAKF;EACE;;;AAIJ;EAEI;IACE","file":"Nav.css"}