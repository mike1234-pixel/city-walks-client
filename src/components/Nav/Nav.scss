/** @format */

.nav-container {
  min-width: 100%;
}
.nav-bar {
  background-color: var(--elegant-color-dark);
  color: white;
  box-shadow: none;
}

.logo {
  font-family: var(--logo-font);
  font-size: 150%;
}

@media (min-width: 991px) {
  .hamburger1 {
    visibility: hidden;
    display: none;
  }
}

.nav-item {
  a[data-test="nav-link"] {
    font-family: var(--display-font);
    &:hover,
    &:focus,
    &:active {
      color: var(--blue) !important;
    }
  }
  .form-control {
    margin-left: 0 !important;
  }
}

.navbar-brand {
  a {
    margin-left: -15px;
  }
}

@media (max-width: 991px) {
  .nav-bar {
    .container {
      padding: 0 15px;
    }
  }
}
