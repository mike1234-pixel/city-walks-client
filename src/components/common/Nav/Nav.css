.nav-container {
  min-width: 100%;
}

.nav-bar {
  background-color: var(--dark);
  color: white;
  box-shadow: none;
  position: sticky;
  top: 0;
  z-index: 20;
  padding: 20px 0;
  transition: all 0.2s;
}
.nav-bar--scroll {
  padding: 0;
}

.logo {
  font-family: var(--logo-font);
  font-size: 150%;
}

@media (min-width: 991px) {
  .hamburger1 {
    visibility: hidden;
    display: none;
  }
}

.nav-item a[data-test="nav-link"] {
  font-family: var(--display-font);
}

.nav-item a[data-test="nav-link"]:hover,
.nav-item a[data-test="nav-link"]:focus,
.nav-item a[data-test="nav-link"]:active {
  color: var(--blue) !important;
}

.nav-item .form-control {
  margin-left: 0 !important;
}

.navbar-brand a {
  margin-left: -15px;
}

@media (max-width: 991px) {
  .nav-bar .container {
    padding: 0 15px;
  }
}
