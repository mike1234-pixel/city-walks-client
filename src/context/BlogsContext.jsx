import { createContext, useState } from "react"

export const BlogsContext = createContext();

export const BlogsContextProvider = (props) => {

    const [blogPosts, setBlogPosts] = useState([])
    const [blogsLoading, setBlogsLoading] = useState(true)

    return (
        <BlogsContext.Provider 
            value={{
                blogPosts,
                setBlogPosts,
                blogsLoading,
                setBlogsLoading,
            }}
        >
            {props.children}
        </BlogsContext.Provider>
    )
}