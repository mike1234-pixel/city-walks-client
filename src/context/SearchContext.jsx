import { createContext, useState } from "react"

export const SearchContext = createContext();

export const SearchContextProvider = (props) => {
    const [searchValue, setSearchValue] = useState("")
    const [redirect, setRedirect] = useState(false)
  
    const handleChangeSearch = (e) => {
      setSearchValue(e.target.value)
    }
  
    const handleSubmitSearch = (e) => {
      e.preventDefault();
      setRedirect(true);
    }
  
    const handleClickSearch = (city) => {
      setSearchValue(city)
      setRedirect(true);
    }

    return (
        <SearchContext.Provider 
            value={{
            handleChangeSearch, 
            handleSubmitSearch,  
            redirect, 
            setRedirect,
            handleClickSearch, 
            searchValue,
            setSearchValue
            }}
        >
            {props.children}
        </SearchContext.Provider>
    )
}