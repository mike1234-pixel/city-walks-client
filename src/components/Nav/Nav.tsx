import React, { useState, useEffect } from "react";
import { useHistory } from "react-router-dom";
import SearchBar from "../SearchBar/SearchBar";
import {
  MDBNavbar,
  MDBNavbarBrand,
  MDBNavbarNav,
  MDBNavItem,
  MDBNavLink,
  MDBHamburgerToggler,
  MDBCollapse,
  MDBContainer,
} from "mdbreact";
import { connect } from "react-redux";
import { Dispatch, bindActionCreators } from "redux";
import { setRedirect } from "../../actions/actions";
import RootState from "../../types/State/Root/State";
import * as Actions from "../../types/Actions";
import "./Nav.scss";

interface Props {
  redirect: boolean;
  setRedirect: (redirect: boolean) => Actions.SetRedirect;
}

const Nav: React.FC<Props> = (props: Props) => {
  const [toggleNav, setToggleNav] = useState<boolean>(false);
  const [scroll, setScroll] = useState<boolean>(false);

  const { redirect, setRedirect } = props;

  const pushSlug: Function = useHistory().push;

  useEffect(() => {
    if (redirect) {
      pushSlug("/walks");
      setRedirect(false);
    }

    window.addEventListener("scroll", () => {
      setScroll(window.scrollY > 2);
    });
  });

  const handleClick: () => void = () => {
    setToggleNav(!toggleNav);
  };

  return (
    <MDBNavbar
      className={scroll ? "nav-bar nav-bar--scroll" : "nav-bar"}
      dark
      expand="lg"
    >
      <MDBContainer>
        <MDBNavbarBrand>
          <span className="white-text">
            <MDBNavLink className="white-text logo" to="/">
              City Walks
            </MDBNavLink>
          </span>
        </MDBNavbarBrand>
        <MDBHamburgerToggler
          color="#fff"
          className="hamburger1"
          id="hamburger1"
          onClick={handleClick}
        />
        <MDBCollapse id="navbarCollapse3" isOpen={toggleNav} navbar>
          <MDBNavbarNav left>
            <MDBNavItem>
              <MDBNavLink className="white-text" to="/">
                Home
              </MDBNavLink>
            </MDBNavItem>
            <MDBNavItem>
              <MDBNavLink
                data-testid="nav-link-cities"
                className="white-text"
                to="/cities"
              >
                Cities
              </MDBNavLink>
            </MDBNavItem>
            <MDBNavItem>
              <MDBNavLink className="white-text" to="/walks">
                Walks
              </MDBNavLink>
            </MDBNavItem>
            <MDBNavItem>
              <MDBNavLink className="white-text" to="/sights">
                Sights
              </MDBNavLink>
            </MDBNavItem>
            <MDBNavItem>
              <MDBNavLink className="white-text" to="/forum">
                Forum
              </MDBNavLink>
            </MDBNavItem>
            <MDBNavItem>
              <MDBNavLink className="white-text" to="/about">
                About
              </MDBNavLink>
            </MDBNavItem>
            <MDBNavItem>
              <MDBNavLink className="white-text" to="/contact">
                Contact
              </MDBNavLink>
            </MDBNavItem>
          </MDBNavbarNav>
          <MDBNavbarNav right>
            <MDBNavItem>
              <SearchBar searchValue={""} />
            </MDBNavItem>
          </MDBNavbarNav>
        </MDBCollapse>
      </MDBContainer>
    </MDBNavbar>
  );
};

const mapStateToProps: (state: RootState) => void = (state) => ({
  redirect: state.searchState.redirect,
});

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    dispatch,
    ...bindActionCreators({ setRedirect }, dispatch),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Nav);
