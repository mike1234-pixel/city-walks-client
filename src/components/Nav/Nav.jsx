import { useState, useEffect, useContext } from "react"
import { useHistory } from 'react-router-dom'
import SearchBar from '../SearchBar/SearchBar'
import { MDBNavbar, MDBNavbarBrand, MDBNavbarNav, MDBNavItem, MDBNavLink, MDBHamburgerToggler, MDBCollapse, MDBContainer, MDBInput } from "mdbreact"
import { SearchContext } from '../../context/SearchContext';
import './Nav.css'

const Nav = () => {

  const [toggleNav, setToggleNav] = useState(false)

  const { redirect, setRedirect } = useContext(SearchContext)

  let history = useHistory()

  // wrapping this in useEffect prevents error: "cannot update a component while rendering a different component"
  useEffect(() => {
    if (redirect) {
      history.push("/walks")
      setRedirect(false)
    }
  })

  const handleClick = () => {
    setToggleNav(!toggleNav)
  }

  return (
      <MDBContainer className="nav-container">
        <MDBNavbar className="nav-bar" dark expand="md">
          <MDBNavbarBrand>
            <b className="white-text"><MDBNavLink className="white-text logo" to="/">City Walks</MDBNavLink></b>
          </MDBNavbarBrand>
          <MDBHamburgerToggler color="#fff" className="hamburger1" id="hamburger1" onClick={handleClick} />
          <MDBCollapse id="navbarCollapse3" isOpen={toggleNav} navbar>
            <MDBNavbarNav left>
              <MDBNavItem>
                <MDBNavLink className="white-text" to="/">Home</MDBNavLink>
              </MDBNavItem>
              <MDBNavItem>
                <MDBNavLink data-testid="nav-link-cities" className="white-text" to="/cities">Cities</MDBNavLink>
              </MDBNavItem>
              <MDBNavItem>
                <MDBNavLink className="white-text" to="/walks">Walks</MDBNavLink>
              </MDBNavItem>
              <MDBNavItem>
                <MDBNavLink className="white-text" to="/blog">Blog</MDBNavLink>
              </MDBNavItem>
              <MDBNavItem>
                <MDBNavLink className="white-text" to="/boards">Forum</MDBNavLink>
              </MDBNavItem>
              <MDBNavItem>
                <MDBNavLink className="white-text" to="/about">About</MDBNavLink>
              </MDBNavItem>
              <MDBNavItem>
                <MDBNavLink className="white-text" to="/contact">Contact</MDBNavLink>
              </MDBNavItem>
            </MDBNavbarNav>
            <MDBNavbarNav right>
              <MDBNavItem>
                <SearchBar/>
              </MDBNavItem>
            </MDBNavbarNav>
          </MDBCollapse>
        </MDBNavbar>
      </MDBContainer>
    )
  }

export default Nav