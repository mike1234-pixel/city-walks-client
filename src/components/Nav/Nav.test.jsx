import renderer from 'react-test-renderer'
import Nav from './Nav'
import { SearchContextProvider } from "../../context/SearchContext"
import { BrowserRouter as Router } from "react-router-dom"

// snapshot test
it('Nav renders correctly', () => {  

  const tree = renderer
    .create(
        <SearchContextProvider>
            <Router>
                <Nav/> 
            </Router>
          </SearchContextProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// Nav contains NavLinks so has to be wrapped in a Router otherwise test throws an error  