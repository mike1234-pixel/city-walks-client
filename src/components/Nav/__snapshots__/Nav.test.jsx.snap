// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
<div
  className="container nav-container"
  data-test="container"
>
  <nav
    className="navbar-dark navbar navbar-expand-md nav-bar"
    data-test="navbar"
    role="navigation"
  >
    <div
      className="navbar-brand"
      data-test="navbar-brand"
    >
      <b
        className="white-text"
      >
        <a
          aria-current="page"
          className="nav-link Ripple-parent white-text logo active"
          data-test="nav-link"
          href="/"
          onClick={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          City Walks
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </a>
      </b>
    </div>
    <input
      className="hamburger-button__checkbox"
      data-test="hamburger-toggler"
      defaultChecked={false}
      id="hamburger1"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="hamburger-button__button hamburger1"
      htmlFor="hamburger1"
      id="nav-icon1"
    >
      <span
        style={
          Object {
            "background": "#fff",
          }
        }
      />
      <span
        style={
          Object {
            "background": "#fff",
          }
        }
      />
      <span
        style={
          Object {
            "background": "#fff",
          }
        }
      />
    </label>
    <div
      className="collapse navbar-collapse"
      data-test="collapse"
      id="navbarCollapse3"
      style={Object {}}
    >
      <ul
        className="navbar-nav mr-auto"
        data-test="navbar-nav"
      >
        <li
          className="nav-item"
          data-test="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link Ripple-parent white-text active"
            data-test="nav-link"
            href="/"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            style={Object {}}
          >
            Home
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </a>
        </li>
        <li
          className="nav-item"
          data-test="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link Ripple-parent white-text"
            data-test="nav-link"
            data-testid="nav-link-cities"
            href="/cities"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            Cities
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </a>
        </li>
        <li
          className="nav-item"
          data-test="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link Ripple-parent white-text"
            data-test="nav-link"
            href="/walks"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            Walks
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </a>
        </li>
        <li
          className="nav-item"
          data-test="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link Ripple-parent white-text"
            data-test="nav-link"
            href="/blog"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            Blog
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </a>
        </li>
        <li
          className="nav-item"
          data-test="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link Ripple-parent white-text"
            data-test="nav-link"
            href="/boards"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            Forum
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </a>
        </li>
        <li
          className="nav-item"
          data-test="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link Ripple-parent white-text"
            data-test="nav-link"
            href="/about"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            About
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </a>
        </li>
        <li
          className="nav-item"
          data-test="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link Ripple-parent white-text"
            data-test="nav-link"
            href="/contact"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            Contact
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </a>
        </li>
      </ul>
      <ul
        className="navbar-nav ml-auto"
        data-test="navbar-nav"
      >
        <li
          className="nav-item"
          data-test="nav-item"
        >
          <form
            onSubmit={[Function]}
          >
            <span
              className="search-container"
              style={
                Object {
                  "marginRight": "40px",
                }
              }
            >
              <div
                className="md-form"
              >
                
                <input
                  aria-disabled={false}
                  className="form-control"
                  data-test="input"
                  data-testid="search-input"
                  disabled={false}
                  id="search-input"
                  name="search-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onInput={[Function]}
                  type="text"
                  value=""
                />
                <label
                  aria-labelledby=""
                  className=""
                  data-error=""
                  data-success=""
                  htmlFor="search-input"
                  id=""
                  onClick={[Function]}
                >
                  Search
                </label>
              </div>
              <button
                className="search-btn"
                data-testid="search-btn"
                type="submit"
              >
                <svg
                  className="search-location-icon"
                  fill="currentColor"
                  height="1em"
                  id="search-btn-icon"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M505.04 442.66l-99.71-99.69c-4.5-4.5-10.6-7-17-7h-16.3c27.6-35.3 44-79.69 44-127.99C416.03 93.09 322.92 0 208.02 0S0 93.09 0 207.98s93.11 207.98 208.02 207.98c48.3 0 92.71-16.4 128.01-44v16.3c0 6.4 2.5 12.5 7 17l99.71 99.69c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.59.1-33.99zm-297.02-90.7c-79.54 0-144-64.34-144-143.98 0-79.53 64.35-143.98 144-143.98 79.54 0 144 64.34 144 143.98 0 79.53-64.35 143.98-144 143.98zm.02-239.96c-40.78 0-73.84 33.05-73.84 73.83 0 32.96 48.26 93.05 66.75 114.86a9.24 9.24 0 0 0 14.18 0c18.49-21.81 66.75-81.89 66.75-114.86 0-40.78-33.06-73.83-73.84-73.83zm0 96c-13.26 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
                  />
                </svg>
              </button>
            </span>
          </form>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
